namerd/core/src/main/scala/io/buoyant/namerd/InterfaceConfig.scala

```diff
@@ -21,6 +26,10 @@ abstract class InterfaceConfig extends PolymorphicConfig {
   )
 
   @JsonIgnore
+  @silent
+  def tlsParams = tls.map(_.params(None, LegacyKeyServerEngineFactory)).getOrElse(Stack.Params.empty)
```

Comment:
It does work, tested with:

```bash
./sbt "namerd-examples/tls:run -log.level=DEBUG -com.twitter.finagle.tracing.debugTrace=true"
```

```bash
./sbt "linkerd-examples/namerd-tls:run -log.level=DEBUG -com.twitter.finagle.tracing.debugTrace=true"
```

However, I've updated mesh interface to use `ApplicationProtocolNames.HTTP_2` and `Netty4ServerEngineFactory()`, and added a test to verfiy `Netty4ServerEngineFactory`.